openvm/src/powdr_extension/plonk/chip.rs

```diff
@@ -206,18 +230,21 @@ impl<'a, F: PrimeField32> PlonkVariables<'a, F> {
         }
     }
 
-    /// Get the value of a variable. None if the value is not known yet.
-    fn get(&self, variable: &Variable<u64>) -> Option<F> {
+    /// Get the value of a variable. None if the variable is temporary but still unknown.
+    fn get(&self, variable: &Variable<AlgebraicReference>) -> Option<F> {
         match variable {
-            Variable::Witness(id) => Some(self.witness[self.column_index_by_poly_id[id]]),
+            Variable::Witness(id) => {
+                Some(self.witness[self.column_index_by_poly_id[&id.poly_id.id]])
+            }
             Variable::Tmp(id) => self.tmp_vars[*id],
-            Variable::Unused => None,
+            // The value of unused cells should not matter.
+            Variable::Unused => Some(F::ZERO),
```

Comment:
Without this change, `derive_tmp_values_for_c` does nothing if columns a or b are unused. When they are unused, the selectors should be such that the values don't influence the result anyway.